lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueAutoPingLockStrategy.java

```diff
@@ -0,0 +1,277 @@
+/*
+ *  Copyright 2016 esbtools Contributors and/or its affiliates.
+ *
+ *  This file is part of esbtools.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.esbtools.eventhandler.lightblue;
+
+import com.redhat.lightblue.client.LightblueException;
+import com.redhat.lightblue.client.Locking;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Uses lightblue's locking APIs with TTL, automatically pinging the lock in the background until it
+ * is released.
+ *
+ * <p>Lightblue locks without TTLs are unsafe. If a client crashes, the lock will not be released.
+ * However, locks with TTLs may expire prematurely. By pinging the lock in a separate thread
+ * periodically, we can substantially lesson the likelihood of unintentionally losing the lock.
+ *
+ * <p>Consumers are expected to check the lock at critical points to ensure it did not expire by
+ * calling {@link LockedResource#ensureAcquiredOrThrow(String)}.
+ */
+public class LightblueAutoPingLockStrategy implements LockStrategy {
```

Comment:
This is somewhat redundant with lightblue client's interface, however, as LockStrategy doc should explain some, having this here gives us flexibility to use something else for locking, which is necessary for sane testing strategies and helps pull out some of the non-trivial configuration of such a strategy out of the repositories' already large constructors. And obviously we naturally get to implement the auto-pinging stuff here.